### ISSUE TITLE
aws_default_vpc failing with Default VPC does not exist

### ISSUE THREAD

**Issue by xorxsan** [hashicorp/terraform-provider-aws#21241]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Hi all,

   I'm having an error when trying to use the `aws_default_vpc` resource.

   According to documentation <https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc>:

> For AWS accounts created after 2013-12-04, each region comes with a Default VPC. This is an advanced resource, and has special caveats to be aware of when using it. Please read this document in its entirety before using this resource.

However, when I try to import this resource, it fails with a `No default VPC found in this region` (I tried several regions).

   I'm not really sure this is a bug on the documentation or on the resource itself.

   Thanks!

### Terraform CLI and Terraform AWS Provider Version

```sh
$ terraform -v
Terraform v1.0.8
on windows_amd64
+ provider registry.terraform.io/hashicorp/aws v3.62.0
```

### Affected Resource(s)

* aws_default_vpc

### Terraform Configuration Files

```hcl
provider "aws" {
  region  = "eu-west-2"
}

resource "aws_default_vpc" "default" {}
```

```hcl
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
    }
  }
  required_version = ">= 0.13"
}
```

Also tried with 
```hcl
resource "aws_default_vpc" "default" {
  tags = {
    Name = "Default VPC"
  }
}
```
### Debug Output

https://gist.github.com/xorxsan/91cc8e28cf3fe627653f35444294ac51

### Panic Output

None

### Expected Behavior

The default vpc for any region should be imported (according to documentation).

### Actual Behavior

The following error was thrown:
```sh
│ Error: No default VPC found in this region.
│
│   with aws_default_vpc.default,
│   on main.tf line 5, in resource "aws_default_vpc" "default":
│    5: resource "aws_default_vpc" "default" {}
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Account was created on 2020.

**Comment by ewbankkit** [comment]
@xorxsan Thanks for raising this issue.
Currently the `aws_default_vpc` resource does not create a default VPC in a region if one does not exist, instead "importing" the existing default VPC for the region - and failing if the default VPC has been deleted.
We plan to alter this behavior in the future - https://github.com/hashicorp/terraform-provider-aws/issues/4959.

**Comment by anGie44** [comment]
Hi @xorxsan , this issue has been addressed with the merge of #22253 (to be released in v4.0 of the provider) such that if no default VPC exists in the current AWS Region, one is now created. Reach out if you have any additional questions!
